azavea/idb-osm-extraction-tool
# Styling pass of app

## Overview
This is a styling pass of the currently available features of the app. As there isn't a ton of time in the contract, these will still not look exactly like the flat designs in abstract.

### Demo
<img width="1792" alt="screen shot 2018-08-10 at 1 02 03 pm" src="https://user-images.githubusercontent.com/5672295/43971204-38408f6e-9c9e-11e8-812e-c0095d26da0c.png">
<img width="1792" alt="screen shot 2018-08-10 at 1 02 07 pm" src="https://user-images.githubusercontent.com/5672295/43971263-6b80b214-9c9e-11e8-85cd-c190a0416c73.png">
<img width="1792" alt="screen shot 2018-08-10 at 1 02 14 pm" src="https://user-images.githubusercontent.com/5672295/43971272-72829dac-9c9e-11e8-8f7c-c9a67cf678bd.png">
<img width="1792" alt="screen shot 2018-08-10 at 1 02 24 pm" src="https://user-images.githubusercontent.com/5672295/43971282-790fe2d8-9c9e-11e8-9057-3ef94b4627a5.png">
<img width="1792" alt="screen shot 2018-08-10 at 1 06 46 pm" src="https://user-images.githubusercontent.com/5672295/43971298-820c9548-9c9e-11e8-8868-c765b345d037.png">

### Notes
- I added an `images/` directory, where some custom icons live (we are pulling in FontAwesome rather than using Fontello on this project) as well as client logos.
- We still need to add the "Sponsored by IDB" message. There is a corresponding image in the `images/` folder.
- We still need (?) to add the "Something look wrong? Update..." message to the bottom-left of the map. This is not dynamic in anyway and is always visible.
- It would be nice to try and style the `react-select` inputs' selected and hover states. @kellyi  and I were discussing how this could be done, but it might require some React-writing rather than SCSS.
- The extract button should be disabled until the "required" fields in the sidebar are filled in.


## Testing Instructions
- `git pull`
- In the browser, put a `.-disabled` class on the `.-extract` button to see the disabled styles
- Try running a search in the Geocoder. Toggle to Coordinates to see the Lat/Long design.
- Click the "i" icon in the sidebar header to see style revisions.
- Click one of the site boundary buttons and try drawing on the map to see the various states of the buttons.
